### ISSUE TITLE
MaxMessageBytes config works only if set in both Kafka and Sarama

### ISSUE THREAD

**Issue by GrigoriyMikhalkin** [Shopify/sarama#1186]
##### Versions

Sarama Version: f21e149e5948e9c9b9034f1a19f430d62e00c4d3
Kafka Version: 2.0.0
Go Version: 1.10

##### Configuration

Producer.MaxMessageBytes = 20000000

##### Problem Description

I'm playing with http server from `example` directory, trying to send file contents(about 20MB) to kafka producer. It seems, that I need to set both `message.max.bytes` param in config for Kafka server and `Producer.MaxMessageBytes` in config for `SyncProducer`. If i'm setting only one, i'm getting error: `kafka server: Message was too large, server rejected it to avoid allocation error.`. Is that expected behavior?

**Comment by varun06** [comment]
Yeah, we also faced same issue recently and that's what it was about. Reading suggest that it is an expected behavior.
